app/models/dataset.rb

```diff
@@ -122,6 +124,46 @@ def complete_publishing
     SendTweetService.new(self).perform
   end
 
+  def deprecated_resource
+    url_deprecated_at.present?
+  end
+
+  def self.check_urls
+    # check if dataset URL is live
+    Dataset.all.each do |dataset|
+      if dataset.url.nil? #dataset URLs can be nil without indicating dead resource,
+        puts "#{dataset.name} lacks URL: #{dataset.url}"
+        Rails.logger.warn "#{dataset.name} lacks URL"
+      end # TODO should this be in eval_response_code?
+
+      if eval_response_code?(dataset.url)
+        puts "#{dataset.name} has URL live at #{dataset.url}"
+        Rails.logger.info "#{dataset.name} live at #{dataset.url}"
+      else
+        puts "#{dataset.name} lacks URL : #{dataset.url}"
+        Rails.logger.warn "#{dataset.name} no longer has a live URL : #{dataset.url}"
+        dataset.update_column(:url_deprecated_at, DateTime.now())
+      end
+    end
+  end
+
+  def self.eval_response_code?(url_string)
```

Comment:
yeah that is a good point - I remember reading stuff like this when working out something related https://github.com/blog/1508-repository-redirects-are-here 
I'm not sure how Octopub currently handles instances where a repo still exists but is renamed or moved to a different user - should this maybe be a separate issue?